npm_package/pink-env/src/index.ts

```diff
@@ -0,0 +1,34 @@
+/** Bytes represented in Uint8Array or hex string */
+type Bytes = Uint8Array | string;
+
+interface ContractCallArgs {
+  /** The callee address */
+  callee: Bytes;
+  /** Gas limit */
+  gasLimit: number | bigint;
+  /** Amount of balance that transfer to the callee */
+  value: number | bigint;
+  /** The selector of the ink message to be called */
+  selector: number;
+  /** Scale encoded input arguments */
+  input: Bytes;
+}
+
+interface DelegateCallArgs {
+  /** The code hash that would delegate to */
+  codeHash: Bytes;
+  /** The selector of the ink message to be called */
+  selector: number;
+  /** Scale encoded input arguments */
+  input: Bytes;
+}
+
+declare global {
+  const pink: {
+    /** Call into a contract */
+    invokeContract(args: ContractCallArgs): Uint8Array;
+    /** Delegate call into a piece of code */
+    invokeContractDelegate(args: DelegateCallArgs): Uint8Array;
```

Comment:
The `__hostCall` is return immediately or it's an async call?

Reply:
It's synced. We don't support any async calls.

Reply:
Got it.